### ISSUE TITLE
Highlight text with links in markdown notation

### ISSUE THREAD

**Issue by felipebayres** [ipikuka/remark-flexible-markers#2]
I´m trying to highlight a text that containts a markdown link in it.

For example `==Google is [quietly ramping upits AI efforts](https://www.wired.com/story/meet-bard-googles-answer-to-chatgpt/) in response to==`

But what really renders is just the string itself: 
![image](https://github.com/ipikuka/remark-flexible-markers/assets/29110284/4e4697f7-612b-4d6d-bc28-4d5fc19518ef)

is this plugin expected to work with links as well?

**Comment by talatkuyuk** [comment]
Thank you for report the issue. I will take care.

**Comment by talatkuyuk** [comment]
Hi @felipebayres,

`remark-flexible-markers` traces over the text nodes, and looks for == signs.

Main problem is this: The markdown link at the example breaks the connection between the leading and trailing signs (==)
 
```markdown
==Google is [here](https://www.google.com) and always available==
```
Above markdown phrase consists 3 nodes of texts:
1st text --> `==Google is `
2nd text (the link has a text as well) --> `here`
3rd text --> ` and always available==`

As you see none of the text nodes does not satisfy the marker format, **==text==**

In order to make connection between the leading and trailing signs (==), due to the fact that there is link that cracks the connection between them, it requires too much effort for the implementation. I need community support for that.

But for now, you can try to put == signs at both sides of each text phrase like this:
```markdown
==Google is ==[==here==](https://www.google.com)== and always available==
```
I am waiting your response.

**Comment by talatkuyuk** [comment]
Fixed with version 1.2.0. See https://github.com/ipikuka/remark-flexible-markers/releases/tag/v1.2.0
